PermissionScope/PermissionConfigs.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  PermissionConfigs.swift
+//  PermissionScope
+//
+//  Created by Nick O'Neill on 8/25/15.
+//  Copyright © 2015 That Thing in Swift. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import Accounts
+
+/**
+*  Protocol for permission configurations.
+*/
+@objc public protocol PermissionConfig {
+    /// Permission type
+    var type: PermissionType { get }
+    /// Message for the body label of the dialog presented when requesting access.
+    var message: String { get }
```

Comment:
can we add the `status()` and `request()` methods every permission has here? 
Then move the methods to their classes? (probably better to split into separate files)